---
title       : Michael Love - Bioc2016
author      : Michael Love
framework   : revealjs
revealjs: # https://github.com/hakimel/reveal.js/
  transition  : linear
  theme       : beige
  center      : "true"
highlighter : highlight.js
hitheme     : tomorrow
widgets     : [mathjax]
mode        : selfcontained
knit        : slidify::knit2slides
---

## Bioconductor Workflows Following Fast, Lightweight RNA-seq Quantifiers

[Michael Love](http://mikelove.github.io) <br>
[@mikelove](http://twitter.com/mikelove) <br>
[Bioc2016](http://bioconductor.org/help/course-materials/2016/BioC2016/) <br>
June 25, 2016 <br>
this talk: http://mikelove.github.io/bioc2016

---

## A typical gene-level RNA-seq pipeline

* Align reads (30 min, 2-5 Gb file)
* Count reads in genes (10-30 min)
* Stats package for inference and EDA
    * DESeq2, edgeR, limma-voom
    * Tried and true: many 3rd party evaluations
    * *Counts* of reads: precision of log2FC

*picture*

---

## Criticisms of the standard count-based pipeline

* Slow, generates large file with exact alignments
* Counts are correlated with feature length
[Trapnell et al 2013](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3869392/)

*picture of DGE, DTE, DTU*

---

## Criticisms of the standard count-based pipeline II

* Discards reads that cannot be uniquely assigned to genes [Robert & Watson 2015](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0734-x)

*picture of easy assignment*

---

## New, fast transcript quantifiers

* Sailfish, Salmon, kallisto
* Not exact alignment, but *pseudo-alignment*
* Set of transcripts of likely origin
* Few min / file, small memory req'd
* Output relative abundance per transcript

---

## Using with gene DE

<http://bioconductor.org/packages/tximport>

* Sum transcript level estimated counts to gene level
* Collapse uncertainty of isoform counts within a gene
* Keep multimapping reads increases sensitivity

*picture of collapsing*

---

## Using with gene DE

* Calculate an offset that accounts for changes in average transcript length

<br>

$$ \textrm{ATL}_{g} \equiv \sum_{i \in g} \theta_{i} \bar{l}_{i}, \quad \sum_{i \in g} \theta_{i} = 1 $$

gene ................................... $g$ <br>
isoform ............................. $i$ <br>
effective length ............ $\bar{l}$ <br>
percent abundance ... $\theta$

---

## Offset is accurate when needed

* We will have most data when the feature length is critical

*picture of unidentifiable*
*picture of long and short*

---

## Ex: Roadmap tissues

[Code on GitHub](https://github.com/mikelove/roadmapSalmon)

Run Salmon on 30 FASTQ: few min / file

```{r eval=FALSE}
# 9 seconds to import and summarize
txi <- tximport(files, type="salmon", tx2gene=tx2gene)
# build DESeq2 object
dds <- DESeqDataSetFromTximport(txi, samps, ~tissue)
# 3 seconds to variance stabilize
vsd <- vst(dds)
plotPCA(vsd, "tissue")
# differential expression
dds <- DESeq(dds)
res <- results(dds)
```

*show PCA picture*

---

## Work in collaboration with

* Charlotte Soneson (UZH)
* Mark Robinson (UZH)

<br>

## Support from

* Rafael Irizarry (DFCI)
* NIH Cancer Training Grant

